lib/cog/command/pipeline/binder.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Cog.Command.Pipeline.Binder do
+  alias Piper.Command.Bind.Scope
+  alias Piper.Command.Bindable
+
+  def bind(unbound_invocation, context) when is_map(context) do
+    scope = Scope.from_map(context)
+    try do
```

Comment:
Seems weird that `Bindable.resolve` can either return an ok/error tuple OR raise an error. Maybe we should change that API to have both `resolve` and `resolve!` with only the latter raising exceptions.